### DISCUSSION TITLE
Loading sentry as a 3rd party script

### DISCUSSION THREAD

**Discussion by jwon-proton** [getsentry/support#95] [Support] [answered]
I'm having trouble finding exactly what I need in the documentation.

I'm wondering what is the suggested, safe way to load sentry and capture exceptions or messages with our own dsn on a site/page that we do not have control of that may already be loading sentry?

Thanks!

**Comment by Kobby-Bawuah** [comment]
Hi @jwon-proton, within the SDK you can reduce the data volume you’re sending. All SDKs provide some mechanism to override if an event should send. Please consult the documentation for your SDK for more information on how to configure this. For example https://docs.sentry.io/platforms/javascript/guides/react/configuration/filtering/#decluttering-sentry

We also have [these options](https://docs.sentry.io/product/accounts/quotas/#inbound-data-filters) to filter on Sentry's servers. Turn on inbound data filters for legacy browsers, browser extensions, localhost, and web crawlers. Any filtered events will not count towards your quota.

**Reply by jwon-proton** [reply]
> Hi @jwon-proton, within the SDK you can reduce the data volume you’re sending. All SDKs provide some mechanism to override if an event should send. Please consult the documentation for your SDK for more information on how to configure this. For example https://docs.sentry.io/platforms/javascript/guides/react/configuration/filtering/#decluttering-sentry
> 
> We also have [these options](https://docs.sentry.io/product/accounts/quotas/#inbound-data-filters) to filter on Sentry's servers. Turn on inbound data filters for legacy browsers, browser extensions, localhost, and web crawlers. Any filtered events will not count towards your quota.

I'm not concerned with data volume.

My situation is that I'm writing a 3rd party script. I have my own dsn to log sentry errors to my own account and dasboard.

But sites that consume my script could possibly also be attempting to load sentry, with their own dsn.

Do you have a mechanism to mitigate this situation? Or will a collision happen, or only the final script fetch/init wins and actually logs errors?

**Comment by Kobby-Bawuah** [comment, marked as answer]
Hello @jwon-proton, 

Thanks for clarifying. Unfortunately, there isn't currently a mechanism that easily addresses this. It's a known issue that we are currently looking at. You can take a look [here](https://github.com/getsentry/sentry-javascript/discussions/5878). 

The third-party situation sometimes causes unpredictable behaviour(collisions). So for now you could turn off the global wrapping, disable integrations, use the try-catch block or use inbound filters.
